pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -467,18 +485,29 @@ func (plugin *kubenetNetworkPlugin) teardown(namespace string, name string, id k
 
 	// The host can choose to not support "legacy" features. The remote
 	// shim doesn't support it (#35457), but the kubelet does.
-	if !plugin.host.SupportsLegacyFeatures() {
-		return utilerrors.NewAggregate(errList)
-	}
-
-	activePodPortMapping, err := plugin.getPodPortMapping()
-	if err == nil {
-		err = plugin.hostportSyncer.SyncHostports(BridgeName, activePodPortMapping)
-	}
-	if err != nil {
-		errList = append(errList, err)
+	if plugin.host.SupportsLegacyFeatures() {
+		activePodPortMapping, err := plugin.getPodPortMapping()
+		if err == nil {
+			err = plugin.hostportSyncer.SyncHostports(BridgeName, activePodPortMapping)
```

Comment:
I suppose it would be bad to break hostport support for all old runtimes that rely on synchostports, but do we have a deprecation plan for this? can you please put that into a TODO (maybe something like "when we've moved hostport support into CNI ...")